presto-main/src/main/java/com/facebook/presto/operator/repartition/OptimizedPartitionedOutputOperator.java

```diff
@@ -561,12 +565,13 @@ private static int getFixedWidthTypeSize(Type type)
 
         private void resetPositions(int positionCount)
         {
-            positions = ensureCapacity(positions, positionCount);
+            positions = ensureCapacity(positions, min(positionCount, (positionCount / partitionCount + 1) * 2));
```

Comment:
@yingsu00 perhaps, add a comment explaining this calculation and maybe consider moving it to the caller; e.g.  have the caller provide an estimate of how many rows there will be in a partition

```
resetPositions(int estimatedPositionCount)
```

Reply:
@mbasmanova Thanks for the suggestion! I updated the PR with the change and added the following comments
```
// We initialize the size of the positions array in each partitionBuffers to be at most the incoming page's positionCount, or roughly two times of positionCount 
// divided by the number of partitions. This is because the latter could be greater than the positionCount when the number of partitions is 1 or positionCount is 1.
```